subgraphs/osmosis/src/modules/Deposit.ts

```diff
@@ -0,0 +1,219 @@
+import { Protobuf } from "as-proto";
+import { BigDecimal, BigInt, cosmos, log } from "@graphprotocol/graph-ts";
+import {
+  Deposit as DepositTransaction,
+  LiquidityPool as LiquidityPoolStore,
+} from "../../generated/schema";
+import {
+  MsgJoinPool,
+  MsgJoinSwapExternAmountIn,
+  MsgJoinSwapShareAmountOut,
+} from "./Decoder";
+import * as constants from "../common/constants";
+import { getOrCreateDexAmmProtocol } from "../common/initializer";
+import * as utils from "../common/utils";
+import { updateMetrics } from "./Metrics";
+
+function createDepositTransaction(
+  from: string,
+  liquidityPool: LiquidityPoolStore,
+  transaction: cosmos.TxResult,
+  block: cosmos.HeaderOnlyBlock,
+  inputTokenAmounts: Array<BigInt>,
+  sharesMinted: BigInt,
+  amountUSD: BigDecimal
+): void {
+  if (!transaction) {
+    return;
+  }
+  const transactionId = "deposit-" + transaction.hash.toHexString();

```

Comment:
Id needs log index!

Reply:
Same reason as the comments above for the swap transaction. Thanks!